ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/ApiSubmissionService.scala

```diff
@@ -78,6 +82,9 @@ class ApiSubmissionService private (
   private val logger = LoggerFactory.getLogger(this.getClass)
   private val validator = TimeModelValidator(timeModel)
 
+  private def isLocal(party: Ref.Party): Future[Boolean] =
+    partyManagementService.listParties().map(_.find(p => p.isLocal && p.party == party).isDefined)
```

Comment:
Not something we should address now, but I remember someone stating that `listParties` could return tens of thousands of entries (likely fetched from a database), so we might have to add some cache here in the future.

Reply:
I would highly recommend pushing the query down to the index instead of adding caches, but I totally get your point.